src/backend/distributed/utils/node_metadata.c

```diff
@@ -82,6 +83,14 @@ master_add_node(PG_FUNCTION_ARGS)
 	Datum returnData = AddNodeMetadata(nodeNameString, nodePort, groupId, nodeRack,
 									   hasMetadata);
 
+	/*
+	 * After adding new node, we also replicate all existing reference tables to the new
+	 * node. ReplicateAllReferenceTablesToAllNodes replicates reference tables to all
+	 * nodes however, it skips nodes which already has healthy placement of particular
+	 * reference table.
+	 */
+	ReplicateAllReferenceTablesToAllNodes();
```

Comment:
Let's find a way not to call this function if the node is already in pg_dist_node.